components/content-service/pkg/storage/gcloud.go

```diff
@@ -889,30 +892,89 @@ func (p *PresignedGCPStorage) DeleteObject(ctx context.Context, bucket string, q
 		err = client.Bucket(bucket).Object(query.Name).Delete(ctx)
 		if err != nil {
 			log.WithField("bucket", bucket).WithField("object", query.Name).Error(err)
+			if err == gcpstorage.ErrBucketNotExist {
+				return ErrNotFound
+			}
 			return err
 		}
 		return nil
 	}
-	if query.Prefix != "" {
-		prefix := query.Prefix
-		if !strings.HasSuffix(prefix, "/") {
-			prefix = prefix + "/"
-		}
 
-		b := client.Bucket(bucket)
-		it := b.Objects(ctx, &storage.Query{
+	prefix := query.Prefix
+	b := client.Bucket(bucket)
+	var it *gcpstorage.ObjectIterator
+	if prefix != "" && prefix != "/" {
+		it = b.Objects(ctx, &storage.Query{
 			Prefix: prefix,
 		})
-		for {
-			attrs, err := it.Next()
-			if err == iterator.Done {
-				break
-			}
-			err = b.Object(attrs.Name).Delete(ctx)
-			if err != nil {
-				log.WithField("bucket", bucket).WithField("object", attrs.Name).Error(err)
-			}
+	} else {
+		it = b.Objects(ctx, nil)
+	}
+	for {
+		attrs, err := it.Next()
+		if err == iterator.Done {
+			break
+		}
+		err = b.Object(attrs.Name).Delete(ctx)
+		if err != nil {
+			log.WithField("bucket", bucket).WithField("object", attrs.Name).Error(err)
 		}
 	}
+
+	if err == gcpstorage.ErrBucketNotExist {
+		return ErrNotFound
+	}
 	return err
 }
+
+// DeleteBucket deletes a bucket
+func (p *PresignedGCPStorage) DeleteBucket(ctx context.Context, bucket string) (err error) {
+	client, err := newGCPClient(ctx, p.config)
+	defer client.Close()
+
+	err = p.DeleteObject(ctx, bucket, &DeleteObjectQuery{})
+	if err != nil {
+		if err == gcpstorage.ErrBucketNotExist {
+			return ErrNotFound
+		}
+		return err
+	}
+
+	// try again, just because the backend is lazy
+	err = p.DeleteObject(ctx, bucket, &DeleteObjectQuery{})
+	if err != nil {
+		if err == gcpstorage.ErrBucketNotExist {
+			return ErrNotFound
+		}
+		return err
+	}
```

Comment:
What does this mean, i.e. why is it neccesary?

Reply:
Don't know if this is really necessary. I took this from here: https://github.com/gitpod-io/gitpod/blob/945041dc6845926bddcbe299f644698cb1e8c003/components/server/src/storage/gcloud-storage-client.ts#L48-L51

Reply:
Looks really odd to me. I'd say we try without rather than maintaining code we don't understand :)